packages/celltags-extension/src/celltag.tsx

```diff
@@ -0,0 +1,204 @@
+
+    if (event.ctrlKey) return;
+
+    if (event.key === 'Enter') {
+      this.addTag(props, input.value);
+    } else if (event.key === 'Escape') {
+      this._emptyAddTag(input);
+    }
+  }
+
+  private _onAddTagFocus(event: React.FocusEvent<HTMLInputElement>) {
+    if (!this._editing) {
+      (event.target as HTMLInputElement).blur();
+    }
+  }
+
+  private _onAddTagBlur(input: HTMLInputElement) {
+    if (this._editing) {
+      this._editing = false;
+      this._emptyAddTag(input);
+    }
+  }
+
+  private _onChange(event: React.ChangeEvent<HTMLInputElement>) {
+    if (!event.target.value) {
+      this._emptyAddTag(event.target);
+    } else {
+      const tmp = document.createElement('span');
+      tmp.className = 'add-tag';
+      tmp.textContent = event.target.value;
+      // set width to the pixel length of the text
+      document.body.appendChild(tmp);
+      event.target.style.width = tmp.getBoundingClientRect().width + 8 + 'px';
+      document.body.removeChild(tmp);
+    }
+  }
+
+  private _onAddTagClick(
+    props: FieldProps,
+    event: React.MouseEvent<HTMLElement>
+  ) {
+    const elem = (event.target as HTMLElement).closest('div');
+    const input = elem?.childNodes[0] as HTMLInputElement;
+    if (!this._editing) {
+      this._editing = true;
+      input.value = '';
+      input.focus();
+    } else if (event.target !== input) {
+      this.addTag(props, input.value);
+    }
+    event.preventDefault();
+  }
+
+  private _onTagClick(props: FieldProps, tag: string) {
+    const data = props.formData;
+    if (data.includes(tag)) {
+      data.splice(data.indexOf(tag), 1);
+    } else {
+      data.push(tag);
+    }
+
+    props.formContext.updateMetadata({ [props.name]: data }, true);
+  }
+
+  render(props: FieldProps): JSX.Element {
+    const allTags: string[] = this.pullTags();
+
+    return (
+      <div className="cell-tags">
+        <div className="jp-FormGroup-fieldLabel jp-FormGroup-contentItem">
+          Cell Tags
+        </div>
+        {allTags &&
+          allTags.map((tag: string, i: number) => (
+            <div
+              key={i}
+              className={`tag ${
+                props.formData.includes(tag) ? 'applied-tag' : 'unapplied-tag'
+              }`}
+              onClick={() => this._onTagClick(props, tag)}
+            >
+              <div className="tag-holder">
+                <span>{tag}</span>
+                {props.formData.includes(tag) && (
+                  <LabIcon.resolveReact
+                    icon={checkIcon}
+                    tag="span"
+                    elementPosition="center"
+                    height="18px"
+                    width="18px"
+                    marginLeft="5px"
+                    marginRight="-3px"
+                  />
+                )}
+              </div>
+            </div>
+          ))}
+        <div className="tag unapplied-tag">
+          <div
+            className="tag-holder"
```

Comment:
Should all caps global const instead of inline CSS class string be used here?

Please see `shell.ts` for reference:
https://github.com/jupyterlab/jupyterlab/blob/82e1a74b61a41fbe5d65e8542053b749cdfe6ea0/packages/application/src/shell.ts#L39